scripts/test-runner.js

```diff
@@ -314,16 +371,42 @@ function runTest(name, code, options, args) {
     return false;
   }
 }
+
+function testInferiorRepl (path) {
+  if (!fs.existsSync(path)) {
+    throw new ArgsParseError(`repl ${path} does not exist`)
+  }
+  // find out how to print
+  let script = `
+    if (typeof (console) !== 'undefined' && console.log !== undefined) {
+      console.log('console.log')
+    }
+    else if (typeof('print') !== 'undefined') {
+      print('print')
+    }`
+  let out = child_process.spawnSync(path, {input: script});
+  let output = String(out.stdout);
+  if (output.trim() === "") {
+    throw new ArgsParseError(`could not figure out how to print in inferior repl ${path}`);
+  }
+  console.log({printName: output.trim(), cmd: path.trim()})
+  return {printName: output.trim(), cmd: path.trim()};
+}
+
 function run(args) {
   let failed = 0;
   let passed = 0;
   let total = 0;
+  if (args.repl !== "") {
+    execSpec = testInferiorRepl(args.repl);
+  }
 
   for (let test of tests) {
     // filter hidden files
     if (path.basename(test.name)[0] === ".") continue;
     if (test.name.endsWith("~")) continue;
     if (test.file.includes("// skip")) continue;
+    if (args.es5 && test.file.includes("// es6")) continue;
```

Comment:
Are you excluding es6 tests from running under es5 mode? I thought you wanted to babel lower es6 tests to es5 code for running?

Reply:
Yes. I am not sure what the right thing to do here is.  I added the babel lowering to deal with test that use ES6 features like let and for-of loops, such they could still run in an es5 context. However some tests explicitly test ES6 features, such as symbols for intstance and testing them in es5 mode is pointless. 
So I ended up having both, but I am open to ideas.

Reply:
I see.
In this situation, it seems wasteful to babel lower for non-es6 tests under es5 mode.
Or if you still want to opt-in some es6 tests for es5 mode, you can introduce two flags: es6-can-lower, es6-cannot-lower(you may change the wording).

Reply:
Why is it wasteful? Seems simpler to me, if you are dealing with an es5-only runtime and the majority of tests was not written with this mindset, lowering them to get rid of syntactical/semantics features seems the cleanest approach allowing you to still use the test. I only added the flag to deal with tests that can't be lowered properly. Would "es6-only" be a better name?

Reply:
The fact that, 90% of the tests are es5-only tests which do not need to be babel lowered; they are all transformed by additional babel lowering API call, is the thing I feel wasteful. 
We can avoid this by introducing three categories: default(no babel call), es6-can-lower(call babel on it), es6-cannot-lower(skip for es5 mode).
But I am fine if you insist on your design since it only slows down test-runner.

Reply:
its a fair point. But your design requires me to go through all 700 or so tests and categorize all of them, which I would like to avoid :).

Reply:
Hmm, I thought you just need to mark es6 tests and decide which can and can't be lowered?
If we want to do it, here are the logic steps in my mind:
1. Directly run the es5 runtime for all the tests. Any failed tests are es6 tests(which I hope are not many). Any succeeded tests are es5 tests do not need babel lowering.
2. Mark these es6 tests with es6-can-lower flag which tells test-runner to babel lower them and run them. Any failed tests are es6 tests that can't be lowered(mark them with es6-cannot-lower flag).
3. Done.

Reply:
Yes. I am not saying it can't be done. I am just saying I think its a lot of work just to avoid some babel invocations. 
If you feel strongly about it, I can do it. But it seems like a lot, for what we are trying achieve here.

Reply:
Yeah, I am fine with this.